test/upgrade/prober/wathola/sender/services.go

```diff
@@ -50,17 +55,27 @@ func (s *sender) SendContinually() {
 		close(shutdownCh)
 	}()
 
+	localRetry := 0
 	for {
 		select {
 		case <-shutdownCh:
+			// If there is ongoing event transition, we need to push to retries too.
+			if localRetry != 0 {
+				s.retries = append(s.retries, localRetry)
+			}
 			return
 		default:
 		}
 		err := s.sendStep()
 		if err != nil {
+			localRetry++
 			log.Warnf("Could not send step event, retry in %v", senderConfig.Cooldown)
 			time.Sleep(senderConfig.Cooldown)
 		} else {
+			if localRetry != 0 {
```

Comment:
Here we should have some configurable threshold, and we should add those retries to report only if `retry >= retriesToReport`

Reply:
hmm, I think this is a good idea to reduce network traffic, Though I don't expect many data point with retry here, it is likely several with long periods. My main concern is that there is no pod log for the sender due to the way our e2e test works, so there will be no way to see how the unavailability behaviour is like unless the test fails.